### DISCUSSION TITLE
specifying the version is required? how to migrate?

### DISCUSSION THREAD

**Discussion by marvinfischenich** [renovatebot/renovate#16539] [Request Help] [answered]
Hi guys,
i want to make use of renovate in our company GitLab.

Is it really required to have the requirement version in the requirements.txt?
Currently most of our repos don't have that.
The requirements.txt looks like that:

```
dependencya
dependencyb
dependencyc
```

From what i understood from the docu it has to look like this for renovate to work:

```
dependencya == 1.2.3
dependencyb == 0.4.6
dependencyc == 1.3
```

Is there a way we can keep our old requirements.txt files?

Or do i have to do a one-time migration in every repo before we can use renovate properly?
If yes what would be the most suitable way of migrating our requirements-files?

**Comment by rarkins** [comment]
What is there to update if there's no version?

**Reply by marvinfischenich** [reply]
the dependency has versions.
we just don't maintain them in our requirements.txt.

so on every MR the latest version will be used.
but if there is no MR for a while no update of the requirement will happen.
that's why we want to introduce renovate.

**Reply by rarkins** [reply]
What type of update (MR) would you expect Renovate to do if the versions are not in requirements.txt?

Renovate's job is to update versions in files in a repo. You say you don't want to have versions in your files in the repo, so what exactly do you think Renovate would do?

**Reply by marvinfischenich** [reply]
i want it but currently i don't have it.
can't renovate just detect the latest version available and put it in the requirement.txt if there is no version specified yet?

**Reply by rarkins** [reply]
In theory that could be done, however we don't have that feature

**Comment by MaronHatoum** [comment]
Renovate extracts versions from manifest files and is dependent on these for update lookup.
checkout the detected dependency section for the example you provided:
https://github.com/MaronHatoum/python-dis/issues/1

**Comment by PhilipAbed** [comment, marked as answer]
1) If `requirements.txt` file has all your direct dependencies without the version, it means everytime you run `pip install -r requirements.txt`
you get latest version. 

2) in case your `requirements.txt` is not maintained or missing dependencies where u installed on your environment but didn't update the `requirements.txt`, then you have to maintain it..

Python works in an Environment, which has all the dependencies, 
if renovate needs to reach those dependencies, it needs to be connected to that environment.

Currently Renovate would take the files from a repository, clone them to the machine where it's running.. but that makes it in another machine
now we don't have any feature which would let us connect to an active python environment where we would be able to run some pip command which would extract the current active dependencies to put them into a `requirements.txt` file

Instead

i would create a daily script(cron job), which updates the `requirements.txt` on that repository,  and then you can run renovate normally
